pkg/volume/iscsi/iscsi_util.go

```diff
@@ -263,66 +254,92 @@ func (util *ISCSIUtil) AttachDisk(b iscsiDiskMounter) error {
 			break
 		}
 	}
-	err = b.mounter.FormatAndMount(devicePath, globalPDPath, b.fsType, nil)
-	if err != nil {
-		glog.Errorf("iscsi: failed to mount iscsi volume %s [%s] to %s, error %v", devicePath, b.fsType, globalPDPath, err)
-	}
 
+	for _, tp := range bkpPortal {
+		// mount it
+		globalPDPath := ""
+		if len(bkpPortal) == 1 {
+			globalPDPath = b.manager.MakeGlobalPDName(*b.iscsiDisk)
+		} else {
+			//
+			globalPDPath = mpathMakeGlobalPDName(*b.iscsiDisk, tp)
+		}
+		notMnt, err := b.mounter.IsLikelyNotMountPoint(globalPDPath)
+		if !notMnt {
+			glog.Infof("iscsi: %s already mounted", globalPDPath)
+			return nil
+		}
+
+		if err := os.MkdirAll(globalPDPath, 0750); err != nil {
+			glog.Errorf("iscsi: failed to mkdir %s, error", globalPDPath)
+			return err
+		}
+		err = b.mounter.FormatAndMount(devicePath, globalPDPath, b.fsType, nil)
+		if err != nil {
+			glog.Errorf("iscsi: failed to mount iscsi volume %s [%s] to %s, error %v", devicePath, b.fsType, globalPDPath, err)
+		}
+	}
 	return err
 }
 
-func (util *ISCSIUtil) DetachDisk(c iscsiDiskUnmounter, mntPath string) error {
-	_, cnt, err := mount.GetDeviceNameFromMount(c.mounter, mntPath)
-	if err != nil {
-		glog.Errorf("iscsi detach disk: failed to get device from mnt: %s\nError: %v", mntPath, err)
-		return err
-	}
-	if err = c.mounter.Unmount(mntPath); err != nil {
-		glog.Errorf("iscsi detach disk: failed to unmount: %s\nError: %v", mntPath, err)
-		return err
-	}
-	cnt--
-	// if device is no longer used, see if need to logout the target
-	if cnt == 0 {
-		device, prefix, err := extractDeviceAndPrefix(mntPath)
+func (util *ISCSIUtil) DetachDisk(c iscsiDiskUnmounter, mntPaths []string) error {
+	var cnt int
+	for _, mntPath := range mntPaths {
+		_, cnt, err := mount.GetDeviceNameFromMount(c.mounter, mntPath)
 		if err != nil {
+			glog.Errorf("iscsi detach disk: failed to get device from mnt: %s\nError: %v", mntPath, err)
 			return err
 		}
-		refCount, err := getDevicePrefixRefCount(c.mounter, prefix)
-		if err == nil && refCount == 0 {
-			// This portal/iqn/iface is no longer referenced, log out.
-			// Extract the portal and iqn from device path.
-			portal, iqn, err := extractPortalAndIqn(device)
+		if err = c.mounter.Unmount(mntPath); err != nil {
```

Comment:
By doing Unmount first and GetDeviceNameFromMount second can't the cnt, cnt-- be skipped
and only use cnt